### DISCUSSION TITLE
Adding lagrange multiplier to the Cahn-Hilliard equation in the split form

### DISCUSSION THREAD

**Discussion by alimostafavi24** [idaholab/moose#27364] [Q&A Modules: Phase field]
Dear MOOSE community,

I added the scalar Lagrange multiplier to enforce a constraint on the volume integral of order parameter. Can someone please explain to me what additional term is added to the original Cahn-Hilliard equation by adding this constraint? Attached is the relevant part of the input file.
@jiangwen84 @GiudGiud 
```
[Variables]
  [./c]
    order = FIRST
    family = LAGRANGE
    [./InitialCondition]
    type = SpecifiedSmoothCircleIC
    variable = c
    x_positions = '0'
    y_positions = '0'
    z_positions = '0'
    radii = '${R}'
    invalue = 1
    outvalue = -1
    int_width = ${epsilon3}
  [../]
[]
  [./w]
    order = FIRST
    family = LAGRANGE
  [../]
  [./lambdac]
    family = SCALAR
    order = FIRST
 [../]
[]

[ScalarKernels]
 [./mean_zero_pressure_c]
    type = AverageValueConstraint
   variable = lambdac
    pp_name = total_solutec
  value =   -4.572052e-03
[]
[]

[Kernels]
  [./mean_zero_c]
  type = ScalarLagrangeMultiplier
  variable = c
   lambda = lambdac
 [../]
  [./c_res]
    type = SplitCHParsed
    variable = c
    f_name = F
    kappa_name = kappa_c
    w = w
  [../]

  [./w_res]
    type =  SplitCHWResAniso #SplitCHWRes
    variable = w
    mob_name = M_c
  [../]
  [./time]
    type = CoupledTimeDerivative
    variable = w
    v = c
  [../]
[]

[Postprocessors]
    [./total_solutec]
    type = ElementIntegralVariablePostprocessor
    variable = c
    execute_on = linear
[]
```

Thanks,

**Comment by GiudGiud** [comment]
Hello

Adding a constraint (that way) is like adding an additional equation that gets solved simultaneously by the nonlinear solver. 
You do not need to change the original equations for that

Having the AverageValueConstraint and the ScalarLagrangeMultiplier kernels looks fine to me

Guillaume

**Reply by alimostafavi24** [reply]
Thanks.

My understanding is that the strong form of the equations remain intact. However, additional term appears in residual term for the order parameter according the (https://github.com/idaholab/large_media/blob/master/framework/scalar_constraint_kernel.pdf). Am I correct?

Regards,

**Reply by GiudGiud** [reply]
if you look at the end of that document, the extra term (9) in the residual is added by the ScalarLagrangeMultiplier kernel which you already have
